### ISSUE TITLE
Joblib silently ignores pickle exceptions and does not use cache

### ISSUE THREAD

**Issue by p-himik** [joblib/joblib#896]
I'm trying to use `memory.cache` on a function that returns a bunch of Pandas named tuples made with `list(dataframe.itertuples())`.
Apparently, such tuples cannot be pickled. I'm fine with that, however I'm not fine with the fact that Joblib doesn't even log the issue. Because of the empty `except` at https://github.com/joblib/joblib/blob/master/joblib/_store_backends.py#L193 the pickle exception is just ignored, and the hint about race condition doesn't really help to debug the issue.

**Comment by tomMoral** [comment]
Indeed, the `except` to catch the directory creation race condition should probably be moved in the `create_location` method. I also run into some race conditions when calling in parallel a cached function that has just been modified. The workers are all trying to delete the directory at once, creating errors. I will try to fix this this week.

**Comment by tomMoral** [comment]
Btw, if you have a reproducing case, I would be very interested, to make sure we solve the problem at hand.

**Comment by p-himik** [comment]
Should be something as simple as
```python
import joblib as jb
import pandas as pd

memory = jb.memory.Memory()

@memory.cache
def f():
  return list(pd.DataFrame({'a': [1]}).itertuples())

f()
```

**Comment by tomMoral** [comment]
Note that if you do not provide an argument `location=` in `Memory`, there will be no caching in `joblib`.

But indeed, this example reproduce the problem.

**Comment by p-himik** [comment]
Right, sorry - I was just reproducing it from my head. In the real code, I have `location` specified.

**Comment by jeremiedbb** [comment]
Fixed by #1359
